### DISCUSSION TITLE
Error sending a get request

### DISCUSSION THREAD

**Discussion by jwhoami** [rest-nvim/rest.nvim#249] [Q&A]
Hi,

I am new to nvim and currently using lazyvim

- I installed rest.nvim
- Created http file with the following contents

GET https://reqres.in/api/users?page=5

- Placed cursor on the get line and sent the request

- I get the following error

E5108: Error executing lua ...nvimRMnCG5/usr/share/nvim/runtime/lua/vim/treesitter.lua:121: There is no parser available for buffer 3 and one could not be created because lang could not be determined. Either pass lang or set the buffer filetype
stack traceback:
	[C]: in function 'error'
	...nvimRMnCG5/usr/share/nvim/runtime/lua/vim/treesitter.lua:121: in function 'get_parser'
	...l/share/nvim/lazy/rest.nvim/lua/rest-nvim/utils/init.lua:218: in function 'read_document_variables'
	...l/share/nvim/lazy/rest.nvim/lua/rest-nvim/utils/init.lua:250: in function 'read_variables'
	...l/share/nvim/lazy/rest.nvim/lua/rest-nvim/utils/init.lua:260: in function 'replace_vars'
	...share/nvim/lazy/rest.nvim/lua/rest-nvim/request/init.lua:249: in function 'parse_url'
	...share/nvim/lazy/rest.nvim/lua/rest-nvim/request/init.lua:282: in function 'get_current_request'
	.../.local/share/nvim/lazy/rest.nvim/lua/rest-nvim/init.lua:25: in function 'run'
	[string ":lua"]:1: in main chunk

What could be the problem?

Thanks

**Comment by jwhoami** [comment]
Reading the message carefully and doing some digging around, the issue with the filetype.

I set the filetype

- :set filetype=http

And it started to work.

Now I need to figure out how to automatically detect filetype

**Comment by dimas-cyriaco** [comment]
If you are using `lazy`, you can do:

```lua
return {
  "rest-nvim/rest.nvim",
  ft = "http",
}
```
